### PR TITLE
[45] Updated GUID fetch logic.

### PR DESCRIPTION
**Pull request by quadraticat**
**Resolves:** [#45](https://github.com/unraid/usb-creator-next/issues/45)

**Changes:**

- Updated the GUID fetch regex to take in any amount of characters, hopefully squashing the Sundisk GUID issue.

**Testing:**

- Unable to test hardware as this laptop doesn't have a USB port, but can check early morning tomorrow (7/29)

**Notes:**

- There seems to be an issue with code signing in the cmake, unable to build project while that section is live. Unsure if this is just a me problem, but the issue persists after multiple installs of the project.

### CODE REVIEW

**Comment by elibosley** [comment]
This looks good - but definitely get the build working and test this before we merge. Try to work with @jlre249 to get your build environment set up fully so that you can start building and testing these changes!

**Comment by quadraticat** [comment]
> This looks good - but definitely get the build working and test this before we merge. Try to work with @jlre249 to get your build environment set up fully so that you can start building and testing these changes!

Build works and tested Sundisk USB and PNY USB.

**Comment by elibosley** [comment]
Awesome! Any chance you can grab me the binaries for Mac / Windows and bump the build version to 1.0.1?

**Comment by elibosley** [comment]
I'd add your .DS_Store file to your gitignore and delete it, as these are basically macos placeholder files.

File: debian/unraid-usb-creator/DEBIAN/control [hunk 1]
```diff
@@ -2,7 +2,7 @@ Package: unraid-usb-creator
 Version: 1.0
 Architecture: amd64
 Maintainer: Floris Bos <bos@je-eigen-domein.nl>
```
Comment (elibosley) [inline debian/unraid-usb-creator/DEBIAN/control:4]: Let's change Maintainer here to Unraid Developers <contact@unraid.net>

File: debian/unraid-usb-creator/usr/bin/unraid-usb-creator
Comment (elibosley) [inline debian/unraid-usb-creator/usr/bin/unraid-usb-creator:1]: I think we should not include these in the repo and prefer to use Github Actions to attach them to each PR / branch.

File: .DS_Store
Comment (elibosley) [inline .DS_Store:1]: Let's delete this file.

File: src/dependencies/drivelist/src/windows/list.cpp [hunk 2]
```diff
@@ -651,7 +651,7 @@ bool GetDeviceVidPidSerialNumber(HDEVINFO hDeviceInfo, PSP_DEVINFO_DATA deviceIn
 
         std::string deviceId_usb{wbuffer};
         std::smatch vid_pid_sn_match;
-        std::regex vid_pid_sn_regex("USB\\\\VID_([0-9A-Za-z]{4})&PID_([0-9A-Za-z]{4})\\\\([0-9A-Za-z]+)");
+        std::regex vid_pid_sn_regex("USB\\\\VID_([0-9A-Za-z]{4})(?:.*)&PID_([0-9A-Za-z]{4})(?:.*)\\\\([0-9A-Za-z]+)");
```
Comment (elibosley) [inline src/dependencies/drivelist/src/windows/list.cpp:654]: Awesome work on this change!

File: src/linux/linuxdrivelist.cpp [hunk 3]
```diff
@@ -61,7 +61,7 @@ namespace Drivelist
         }
 
 
-        QRegularExpression vidRegExp("VENDOR_ID=([0-9A-Za-z]{4})"), pidRegExp("MODEL_ID=([0-9A-Za-z]{4})"), snRegExp("SERIAL_SHORT=([0-9A-Za-z]+)");
+        QRegularExpression vidRegExp("VENDOR_ID=([0-9A-Za-z]{4})(?:.*)"), pidRegExp("MODEL_ID=([0-9A-Za-z]{4})(?:.*)"), snRegExp("SERIAL_SHORT=([0-9A-Za-z]+)");
```
Comment (elibosley) [inline src/linux/linuxdrivelist.cpp:64]: Wow all of these are so different.

**Pull request merged**